mesonbuild/build.py

```diff
@@ -697,13 +697,22 @@ class BuildTarget(Target):
 
     install_dir: T.List[T.Union[str, Literal[False]]]
 
-    def __init__(self, name: str, subdir: str, subproject: SubProject, for_machine: MachineChoice,
-                 sources: T.List['SourceOutputs'], structured_sources: T.Optional[StructuredSources],
-                 objects, environment: environment.Environment, compilers: T.Dict[str, 'Compiler'], kwargs):
+    def __init__(
+            self,
+            name: str,
+            subdir: str,
+            subproject: SubProject,
+            for_machine: MachineChoice,
+            sources: T.List['SourceOutputs'],
+            structured_sources: T.Optional[StructuredSources],
+            objects: T.List[T.Union[str, File, ExtractedObjects, GeneratedTypes]],
```

Comment:
I wonder if it's practical in this commit, to instead define the objects as `objects: TargetObjects` or something, defined at the top of the file. It's a bit repetitive, and also the main reason 3 lines becomes 11.

Reply:
Also the commit message here says "mmissing"

Reply:
Yeah, that makes sense. Originally this was `str | File | ExtractedObjects`, which wasn't too bad, but with GeneratedTypes it's a little verbose